substrate/frame/support/procedural/src/pallet/expand/hooks.rs

```diff
@@ -186,6 +201,32 @@ pub fn expand_hooks(def: &mut Def) -> proc_macro2::TokenStream {
 			}
 		}
 
+		impl<#type_impl_gen>
+			#frame_support::traits::BeforeAllRuntimeMigrations
+			for #pallet_ident<#type_use_gen> #where_clause
+		{
+			fn before_all_runtime_migrations() -> #frame_support::weights::Weight {
+				use #frame_support::traits::Get;
+				use #frame_support::__private::hashing::twox_128;
+				use #frame_support::storage::unhashed::contains_prefixed_key;
+				#frame_support::__private::sp_tracing::enter_span!(
```

Comment:
I was at some point skeptical if these tracing spans are making anyone happy, or if we should keep them. Something to keep in mind, but out of scope for this PR.